pallets/exchange/src/lib.rs

```diff
@@ -390,9 +392,13 @@ impl<T: Config> Pallet<T> {
 			let mut bvec = Vec::<Intention<T>>::new();
 			let mut total = 0;
 
+			// REVIEW: I think you can avoid the allocation of the vec here if you use `take_while`
+			// https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.take_while
+			// on an iterator of the slice.
```

Comment:
Thanks. 
I finally refactored the matching part of the algorithm - using also your take_while suggestion. Also removed the unnecessary vec copy.

[PR Here](https://github.com/galacticcouncil/Basilisk-node/blob/cf67165c6eca7fdccc1b48f630ed4542d70849f9/pallets/exchange/src/lib.rs#L389)